caldav/davclient.py

```diff
@@ -36,13 +36,15 @@ class DAVResponse:
     tree = None
     headers = {}
     status = 0
+    huge_tree = False
 
-    def __init__(self, response):
+    def __init__(self, response, huge_tree):
```

Comment:
huge_tree should be defaulted to False.  Also, ref my comment on #300, maybe a more "future proof" way of doing this is to pass the DAVClient object rather than huge_tree.

Reply:
Yes I noticed that from the last failed CI build.
I will rewrite that part and pass the DAVClient object, it will be easier to maintain and add new future options.

Reply:
Any progress?  Do you want me to take over?

Reply:
What do you think about this version ? Pro is it doesn't break anything if someone uses DAVRequest directly. If you want to tweak some changes, you can go ahead, I will do it otherwise.

Reply:
Looks good.  (I dislike a bit that the huge_tree-attribute is stored doubly up in the DAVResponse class, but ... yes, backward compatibility is important and this seems to be the simplest solution to that).

Unit tests or functional test would be good.

Reply:
I added a test, I don't like to add a 10MB file for that, maybe we can autogenerate a random content..
A bout the double huge_tree attribute, I didn't want to duplicate some logic code into XMLParser callbut I can add a function has_huge_tree in DAVResponse, so we dont have to store it twice, do you want me to do that ?

Reply:
About autogeneration ... it doesn't need to be so very random, some silly code like this would probably do:

```
event.icalendar_component['DESCRIPTION'] = "\n".join(["very long description - this is line %i" % x for x in range(0,1234567)
```

... or even ...

```
event.icalendar_component['DESCRIPTION'] = "This is a very long description\n" * 1234567
```

as for `self.has_huge_tree()` vs `self.huge_tree` ... do whatever seems easiest/best for you.